examples.txt

```diff
@@ -23,6 +23,7 @@ Goroutines
 Channels
 Channel Buffering
 Channel Synchronization
+Waiting For Goroutines To Finish
```

Comment:
If there's a word or small phrase in Go for the concept covered by an example I tend to use that, even if that word or phrase isn't intelligible on its own. E.g. "Select" instead of "Waiting on multiple channel operations". Sometimes there isn't a word in go, e.g. for "Non-Blocking Channel Operations". So here I'd favor title "WaitGroups" and slug `/waitgroups`.

Reply:
I'm not sure where this example should go. This is one possibility. Another would be before or after the Worker Pool example (they seem related)? Indeed I'm curious to hear your thoughts on the question I linked from the corresponding issue another user filed about Worker Pools and WaitGroups.

Reply:
Alright, moving to `waitgroup`. Named it in singular to match the type, similarly to `Select` (not `Selects`), but if you feel that `WaitGroups` is better than `WaitGroup`, LMK. Also moving it to after the worker pool, which will mention it

Reply:
Yes let's use the plural. Good observation on select, bu the slightly more nuanced convention is singlural for keyword syntax (if, else, switch, select) and plural for entities (maps, arrays, tickers, worker pools).